stagemonitor-requestmonitor/src/main/java/org/stagemonitor/requestmonitor/RequestMonitor.java

```diff
@@ -186,6 +191,33 @@ private RequestMonitor(Configuration configuration, Metric2Registry registry, Re
 		}
 	}
 
+	public <T extends RequestTrace> ListenableFuture<RequestInformation<T>> monitorAsync(MonitoredRequest<T> monitoredRequest) {
```

Comment:
Same as in `MonitoredRequest`: maybe it would make sense to refactor `monitor` to be able to handle async requests as well as sequential requests.